gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java

```diff
@@ -81,6 +84,14 @@ public StorageObject apply(BlobInfo blobInfo) {
   private final String contentLanguage;
   private final Integer componentCount;
 
+  public static final class ImmutableEmptyMap<K, V> extends AbstractMap<K, V> {
```

Comment:
Though class must be public (and publicly constructable) we should warn users from using it directly.

Lets add a javadoc suggesting that this class should be considered "internal" and therefore should not be used directly and that it can be removed in the future.

If you want to take a more extreme measure, we could create an "internal" sub package and add a filter to the javadoc generation to exclude it.